app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt

```diff
@@ -8,31 +8,47 @@ package mozilla.lockbox.presenter
 
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.addTo
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.components.service.fxa.sharing.ShareableAccount
+import mozilla.lockbox.action.AccountAction
 import mozilla.lockbox.action.AppWebPageAction
 import mozilla.lockbox.action.DialogAction
 import mozilla.lockbox.action.RouteAction
 import mozilla.lockbox.flux.Dispatcher
 import mozilla.lockbox.flux.Presenter
+import mozilla.lockbox.store.AccountStore
 import mozilla.lockbox.store.FingerprintStore
 
 interface WelcomeView {
-    val getStartedClicks: Observable<Unit>
+    val getStartedAutomaticallyClicks: Observable<Unit>
+    val getStartedManuallyClicks: Observable<Unit>
     val learnMoreClicks: Observable<Unit>
+    fun showExistingAccount(email: String)
+    fun hideExistingAccount()
 }
 
+@ExperimentalCoroutinesApi
 class WelcomePresenter(
     private val view: WelcomeView,
     private val dispatcher: Dispatcher = Dispatcher.shared,
+    private val accountStore: AccountStore = AccountStore.shared,
     private val fingerprintStore: FingerprintStore = FingerprintStore.shared
 ) : Presenter() {
+
     override fun onViewReady() {
-        view.getStartedClicks
-            .map {
-                if (fingerprintStore.isKeyguardDeviceSecure)
-                    RouteAction.Login
-                else {
-                    DialogAction.OnboardingSecurityDialog
+        accountStore.shareableAccount()?.let { account ->
```

Comment:
Not sure if I like the `.map{}.subscribe{}` inside of the `.let{}` here

Reply:
Think of this as an 

```kotlin
val account = accountStore.shareableAccount()
if (account != null) {
   // …
}
```

but safer.

Happy to adopt another suggestion :)